### PR TITLE
Remove `TestResponse` class from pytest collecting

### LINKED ISSUE #2317: TestResponse class gets collected by pytest
When `TestResponse` is imported in a function with pytest (i.e. for type-checking annotations), the `TestResponse` class is collected by pytest and attempted to be run as a test suite because it's naming starts with `Test`, resulting in this warning:

```
../usr/local/lib/python3.9/site-packages/werkzeug/test.py:1263
  /usr/local/lib/python3.9/site-packages/werkzeug/test.py:1263: PytestCollectionWarning: cannot collect test class 'TestResponse' because it has a __init__ constructor (from: tests/test_example.py)
    class TestResponse(Response):
```

To replicate the bug, create a new file `tests/test_example.py` with the line below and run `pytest tests/test_example.py`
```
from werkzeug.test import TestResponse
```

To avoid this, there could be the `__test__ = False` flag on the `TestResponse` class (see [here](https://adamj.eu/tech/2020/07/28/how-to-fix-a-pytest-collection-warning-about-web-tests-test-app-class/))

Environment:

- Python version: 3.9.6
- Werkzeug version: 2.0.1

### PR DESCRIPTION
**Pull request by Yourun-proger**
- fixes #2317 

Checklist:

- [ ] Add tests that demonstrate the correct behavior of the change. Tests should fail without the change.
- [ ] Add or update relevant docs, in the docs folder and in code.
- [ ] Add `.. versionchanged::` entries in any relevant code docs.
- [x] Run `pre-commit` hooks and fix any issues.
- [x] Run `pytest` and `tox`, no tests failed.

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -16,6 +16,7 @@ Unreleased
     :issue:`2188`
 -   Redirects with the test client set ``RAW_URI`` and ``REQUEST_URI``
     correctly. :issue:`2151`
+-   ``TestResponse`is no longer collected by ``pytest``. :issue:`2317`
```
Comment (ThiefMaster) [inline CHANGES.rst:19]: I don't think this needs a changelog entry; it's not relevant for people using werkzeug

Comment (Yourun-proger) [reply CHANGES.rst:19]: Understood

Comment (Yourun-proger) [reply CHANGES.rst]: Done!

**Comment by Yourun-proger** [comment]
Why did the test for Windows fall, while others did not, although everything is the same there? Should I pay attention to this?

**Pull request merged**